### ISSUE TITLE
[FAILED] Failed to mount /sysroot; yields never ending loop of udev/journal restart jobs

### ISSUE THREAD

**Issue by FirefighterBlu3** [coreos/bugs#379]
This is against the 2015-jun-10 -stable images for doing a PXE boot.

```syslog
Accepted new private connection.
ConditionPathExists=/etc/initrd-release succeeded for initrd-root-fs.target.
initrd-root-fs.target changed dead -> active
Job initrd-root-fs.target/start finished, result=done
ConditionPathExists=/etc/initrd-release succeeded for initrd-parse-etc.service.
[    6.285191] systemd-udevd[217]: device 0x7f771a8dc2a0 has devpath '/kernel/slab'
[    6.285366] systemd-udevd[217]: seq 648 running
About to execute: /usr/bin/systemctl daemon-reload
[    6.285553] systemd-udevd[217]: passed -1 bytes to netlink monitor 0x7f771a8e4380
Forked /usr/bin/systemctl as 367
[    6.285734] systemd-udevd[217]: passed -1 bytes to netlink monitor 0x7f771a8df8f0
initrd-parse-etc.service changed dead -> start-pre
sysroot.mount: Directory /sysroot to mount over is not empty, mounting anyway.
sysroot.mount failed to run 'mount' task: No such file or directory
sysroot.mount changed dead -> failed
Job sysroot.mount/start finished, result=failed
[FAILED] Failed to mount /sysroot.
See "systemctl status sysroot.mount" for details.
[    6.285927] systemd-udevd[217]: seq 649 processed with 0
Unit sysroot.mount entered failed state.
         Mounting /sysroot...
```

a few more job start/finish, often with ENOENTs, and then a neverending cycle of attempts to [re]start udev Kernel Device Manager followed by attempts to restart Journal Service. one presumes these failures due to the earlier failure of sysroot.mount.

**Comment by FirefighterBlu3** [comment]
on rd.break=pre-mount, if i manually unmount /sysroot/usr (which is usr.squashfs), then rmdir /sysroot/usr, then exit the dracut shell, coreos-stable (2015-06-10) will boot perfectly for the remainder.

```
pre-mount:/# mount
[...]
/root.btrfs on /sysroot type btrfs (rw,relatime,compress=lzo,discard,space_cache)
/usr.squashfs on /sysroot/usr type squashfs (ro,relatime)
pre-mount:/# umount /sysroot/usr
pre-mount:/# rmdir /sysroot/usr
pre-mount:/# ls -al /sysroot
total 4
drwxr-xr-x  1 root 0 0 Jun 11 06:42 .
drwxr-xr-x 13 root 0 0 Jun 11 06:41 ..
pre-mount:/# exit
```

exit the emergency shell twice, and you'll quickly reach a normal bootup.

```
localhost login: core (automatic login)

CoreOS stable (647.2.0)
Update Strategy: No Reboots
core@localhost ~ $
```

**Comment by FirefighterBlu3** [comment]
perhaps there needs to be an ```rmdir /sysroot/usr``` after sysroot-usr.mount is stopped.

```
pre-mount:/# systemctl stop sysroot-usr.mount
[  OK  ] Stopped Reload[   96.684058] systemd-journald[117]: Failed to create new system journal: No such file or directory
 Configuration from the Real Root.
[  OK  ] Reached target Initrd File Systems.
[  OK  ] Stopped target Initrd Root File System.
         Unmounting /sysroot/usr...
[  OK  ] Unmounted /sysroot/usr.
```

**Comment by crawford** [comment]
@FirefighterBlu3 is this issue reproducible? When you first enter the emergency shell, can you run `ls -laR /sysroot`?

**Comment by FirefighterBlu3** [comment]
100% reproducible and yes.
```
pre-mount:/# ls -laR /sysroot|sed -n '$='
8220
```

**Comment by FirefighterBlu3** [comment]
```
pre-mount:/# ls -laR /sysroot|sed 20q   
/sysroot:
total 4
drwxr-xr-x  1 root 0   6 Jun 11 22:11 .
drwxr-xr-x 13 root 0   0 Jun 11 22:11 ..
drwxr-xr-x 10 root 0 225 May 27 03:51 usr

/sysroot/usr:
total 4
drwxr-xr-x 10 root 0  225 May 27 03:51 .
drwxr-xr-x  1 root 0    6 Jun 11 22:11 ..
-r--r--r--  1 root 0    0 May 27 03:57 .noupdate
drwxr-xr-x  3 root 0 7927 May 27 03:51 bin
drwxr-xr-x  4 root 0  137 May 27 03:51 boot
lrwxrwxrwx  1 root 0    5 May 27 03:46 lib -> lib64
drwxr-xr-x 39 root 0 9940 May 27 03:51 lib64
drwxr-xr-x  6 root 0  367 May 27 03:50 libexec
drwxr-xr-x  5 root 0   62 May 27 03:46 local
drwx------  2 root 0    3 May 27 03:45 lost+found
lrwxrwxrwx  1 root 0   20 May 27 03:51 postinst -> sbin/coreos-postinst
drwxr-xr-x  2 root 0 4851 May 27 03:51 sbin
```

**Comment by crawford** [comment]
I assume you are PXE booting this image. Can you tell me the command line you are using?

**Comment by FirefighterBlu3** [comment]
yes and here:

```
   MENU LABEL Boot ^CoreOS Image
   kernel CoreOS/coreos_production_pxe.vmlinuz
   initrd CoreOS/coreos_production_pxe_image.cpio.gz
   APPEND console=ttyS0,115200 edd=off rootfstype=btrfs coreos.autologin=ttyS0 rd.luks=0 rd.lvm=0 rd.md=0 rd.dm=0 LOGLEVEL=1 quiet rd.break=pre-mount
```

**Comment by FirefighterBlu3** [comment]
nope, the freshly fetched stable images produce the same effect.

**Comment by crawford** [comment]
From what URL were you downloading? What are the two versions with which you've tested at this point?

**Comment by FirefighterBlu3** [comment]
those listed in the stable channel at https://coreos.com/docs/running-coreos/bare-metal/booting-with-pxe/

current version appears to be 681.0.0, previous version was 647.2.0

**Comment by FirefighterBlu3** [comment]
i'd point out that after doing an install to disk (v647.2.0), the boot is not broken. it is only the PXE boot that fails

**Comment by mischief** [comment]
it seems that removing `rootfstype=btrfs` allows the system to boot successfully. i'm still investigating.

**Comment by mischief** [comment]
@FirefighterBlu3 i believe this is fixed now, and the change should appear in the next alpha later this week. please try it when it is released.

**Comment by crawford** [comment]
This has been fixed in 717.0.0. Please reopen if you still have trouble.
